app/models/concerns/guide_container.rb

```diff
@@ -29,6 +29,18 @@ def timed?
     false
   end
 
+  def capped?
+    false
```

Comment:
Actually we should consider here that an exercise may be capped or not based depending on its type - e.g. you can limit attempts for choices but not for problems.